packages/victory-core/src/victory-util/textsize.ts

```diff
@@ -243,6 +243,81 @@ const _approximateTextHeightInternal = (text: string | string[], style) => {
   }, 0);
 };
 
+const _approximateDimensionsInternal = (
+  text: string | string[],
+  style?: TextSizeStyleInterface,
+) => {
+  const angle = Array.isArray(style)
+    ? style[0] && style[0].angle
+    : style && style.angle;
+  const height = _approximateTextHeightInternal(text, style);
+  const width = _approximateTextWidthInternal(text, style);
+  const widthWithRotate = angle
+    ? _getSizeWithRotate(width, height, angle)
+    : width;
+  const heightWithRotate = angle
+    ? _getSizeWithRotate(height, width, angle)
+    : height;
+  return {
+    width: widthWithRotate,
+    height: heightWithRotate * coefficients.heightOverlapCoef,
+  };
+};
+
+const _getMeasurementContainer = memoize(() => {
+  const element = document.createElementNS("http://www.w3.org/2000/svg", "svg");
+  element.setAttribute("xlink", "http://www.w3.org/1999/xlink");
+
+  const containerElement = document.createElementNS(
+    "http://www.w3.org/2000/svg",
+    "text",
+  );
+  element.appendChild(containerElement);
+
+  element.style.position = "fixed";
+  element.style.top = "-9999px";
+  element.style.left = "-9999px";
+
+  document.body.appendChild(element);
+
+  return containerElement;
+});
+
+const _measureDimensionsInternal = (
```

Comment:
I'd like to bring up performance...
Currently, it looks like we are calling `approximateTextSize` inside the render method of `VictoryLabel`.  This could certainly be problematic, considering we're doing DOM manipulation in the method now.  And every re-render (eg. animation) will have to execute all this logic repeatedly, even though the dimensions are unlikely to change.

I'd like to improve this approach, if possible.  Some ideas that come to mind:
1. On initial render, use the old `approximateTextSize` (using pre-defined glyph measurements), then inside a `useLayoutEffect` we could perform these measurements, to get a higher accuracy
2. We should memoize these calculations somewhere.  Probably using `useMemo` inside `VictoryLabel`
3. Is the `approximateTextSize` logic able to know whether it's accurate?  Like, if we're measuring unicode symbols that aren't defined, or if the font face is undefined, then we know that the "approximation" will be wrong?   If so, we could ONLY perform the "measure" logic if we think the approximation is wrong.  Not sure if it's totally worth it, but something to think about.

Reply:
I can certainly look into avoiding expensive re-calculations. `useMemo` is likely to be our best bet. 

Since `useLayoutEffect` blocks render until the layout effect has been applied, where we'd then do a setState and cause a rerender, this idea would pretty much just make the initial render even slower.

I'll do some investigating and measuring and see if there's anything clever I can do.

Reply:
Yeah `useLayoutEffect` is probably the wrong place, maybe `useEffect` but that could cause a CLS.  
I guess that's why I was thinking "#3" above would help, so that we could avoid measurements completely, for most common scenarios.

Reply:
Unfortunately, I don't think 3 is going to be reliable. It's possible to get false positives, where we think we know we have the correct font-family etc. but when we render, the used font differs from the one we though would be used. I mention as much in the PR description, but essentially there are various ways that the font-family we're using for our computation would be different than the rendered font.

I have added memoization to the measurement function, so repeated measurements of the same label/tooltip/whatever text now comes for free.

Reply:
Ok, sounds good. 
I looked at the "approximate" logic and yes, it's hard to improve it. The algorithm that the browser uses to render text is impossible to recreate in user land. Measuring is the only reliable way, even though it's quite expensive.